x-pack/plugins/apm/public/index.ts

```diff
@@ -24,6 +24,7 @@ export interface ConfigSchema {
     migrationToFleetAvailable: boolean;
     sourcemapApiAvailable: boolean;
     storageExplorerAvailable: boolean;
+    profilingIntegrationAvailable: boolean;
```

Comment:
The mock will also need the flag here: https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/public/context/apm_plugin/mock_apm_plugin_context.tsx/#L80